docs/AI-for-security/knowledge-base.asciidoc

```diff
@@ -0,0 +1,139 @@
+
+[discrete]
+[[enable-knowledge-base]]
+== Enable knowledge base
+
+There are two ways to enable knowledge base.
+
+NOTE: You must individually enable knowledge base for each {kib} space where you want to use it.
+
+[discrete]
+=== Option 1 — enable knowledge base from an AI Assistant conversation
+
+Open a conversation with AI Assistant, select a large language model, then click **Setup Knowledge Base**. If the button doesn't appear, knowledge base is already enabled.
+
+image::images/knowledge-base-assistant-setup-button.png[An AI Assistant conversation showing the Setup Knowledge Base button]
+
+Knowledge base setup may take several minutes. It will continue in the background if you close the conversation. After setup is complete, you can access knowledge base settings from the conversation settings menu.
+
+image::images/knowledge-base-assistant-menu-dropdown.png[AI Assistant's dropdown menu with the knowledge base option highlighted]
+
+[discrete]
+=== Option 2 — enable knowledge base from the Security AI settings menu
+
+. To open Security AI settings, use the {kibana-ref}/introduction.html#kibana-navigation-search[global search field] to find "AI assistant for security".
+. Click **Setup Knowledge Base**. If the button doesn't appear, knowledge base is already enabled.
+
+image::images/knowledge-base-assistant-settings-kb-tab.png[AI Assistant's settings menu open to the knowledge base tab]
+
+[discrete]
+[[rag-for-alerts]]
+== Knowledge base for alerts
+When knowledge base is enabled, AI Assistant receives alerts from your environment as context for each of your prompts. It will receive alerts from the last 24 hours that have a status of `open` or `acknowledged`, ordered first by risk score, then by recency. Building block alerts are excluded. This enables it to answer questions about multiple alerts in your environment, rather than just the individual alerts you choose to include as context. 
+
+To enable knowledge base for alerts:
+
+. Make sure that knowledge base is <<enable-knowledge-base, enabled>>.
+. Use the slider on the Security AI settings menu's Knowledge Base tab to select the number of alerts to send to AI Assistant. Click **Save**.
+
+NOTE: Including a large number of alerts may cause your request to exceed the maximum token length of your third-party generative AI provider. If this happens, try selecting a lower number of alerts to send.
+
+[discrete]
+[[knowledge-base-add-knowledge]]
+== Add knowledge 
+
+To view all knowledge base entries, go to the Security AI settings menu's Knowledge Base tab. You can add either individual documents, or entire indices containing multiple documents. Each entry in the knowledge base (a document or index) has a **Sharing** setting of either `private` or `global`. Private entries do not affect other users in the {kib} space, whereas global entries affect all users. Each entry can also be `Required knowledge`, which means it will be included as context to every message to AI Assistant. 
```

Comment:
```suggestion
To view all knowledge base entries, go to the Security AI settings and select the **Knowledge Base** tab. You can add individual documents or entire indices containing multiple documents. Each entry in the Knowledge Base (a document or index) has a **Sharing** setting of `private` or `global`. Private entries apply to the current user only and do not affect other users in the {kib} space, whereas global entries affect all users. Each entry can also have a `Required knowledge` setting, which means it will be included as context for every message sent to AI Assistant. 
```